src/main/java/qna/domain/Question.java

```diff
@@ -1,95 +1,106 @@
-    private String title;
 
-    @Lob
-    private String contents;
+  @Column(length = 100, nullable = false)
+  private String title;
 
-    @ManyToOne
-    @JoinColumn(foreignKey = @ForeignKey(name = "fk_question_writer"))
-    private User writer;
+  @Lob
+  private String contents;
 
-    @OneToMany(mappedBy = "question", cascade = CascadeType.ALL)
-    @Where(clause = "deleted = false")
-    @OrderBy("id ASC")
-    private List<Answer> answers = new ArrayList<>();
+  @ManyToOne
+  @JoinColumn(foreignKey = @ForeignKey(name = "fk_question_writer"))
+  private User writer;
 
-    private boolean deleted = false;
+  @Embedded
+  private final Answers answers = new Answers();
 
-    public Question() {
-    }
+  private boolean deleted = false;
 
-    public Question(String title, String contents) {
-        this.title = title;
-        this.contents = contents;
-    }
+  public Question() {
+  }
 
-    public Question(long id, String title, String contents) {
-        super(id);
-        this.title = title;
-        this.contents = contents;
-    }
+  public Question(String title, String contents) {
+    this.title = title;
+    this.contents = contents;
+  }
 
-    public String getTitle() {
-        return title;
-    }
+  public Question(long id, String title, String contents) {
+    super(id);
+    this.title = title;
+    this.contents = contents;
+  }
 
-    public Question setTitle(String title) {
-        this.title = title;
-        return this;
-    }
+  public String getTitle() {
+    return title;
+  }
 
-    public String getContents() {
-        return contents;
-    }
+  public Question setTitle(String title) {
+    this.title = title;
+    return this;
+  }
 
-    public Question setContents(String contents) {
-        this.contents = contents;
-        return this;
-    }
+  public String getContents() {
+    return contents;
+  }
 
-    public User getWriter() {
-        return writer;
-    }
+  public Question setContents(String contents) {
+    this.contents = contents;
+    return this;
+  }
 
-    public Question writeBy(User loginUser) {
-        this.writer = loginUser;
-        return this;
-    }
+  public User getWriter() {
+    return writer;
+  }
 
-    public void addAnswer(Answer answer) {
-        answer.toQuestion(this);
-        answers.add(answer);
-    }
+  public Question writeBy(User loginUser) {
+    this.writer = loginUser;
+    return this;
+  }
 
-    public boolean isOwner(User loginUser) {
-        return writer.equals(loginUser);
-    }
+  public void addAnswer(Answer answer) {
+    answer.toQuestion(this);
+    answers.answers().add(answer);
+  }
 
-    public Question setDeleted(boolean deleted) {
-        this.deleted = deleted;
-        return this;
-    }
+  private boolean isOwner(User loginUser) {
+    return writer.equals(loginUser);
+  }
 
-    public boolean isDeleted() {
-        return deleted;
-    }
+  private Question setDeleted(boolean deleted) {
```

Comment:
개인적인 의견을 드리자면 `set`으로 값을 변경하기 보다는 전적으로 
`Question`의 역할과 책임으로 만드는게 더 좋을 것 같습니다 😄 
```suggestion
  private Question markDeleted() {
```

이 부분은 참고만 부탁 드립니다 😄